types/wicg-file-system-access/index.d.ts

```diff
@@ -91,7 +91,7 @@ interface WritableStream {
     close(): Promise<void>;
 }
 
-class FileSystemWritableFileStream extends WritableStream {
+interface FileSystemWritableFileStream extends WritableStream {
```

Comment:
Those are potentially breaking changes because checks like `typeof FileSystemWritableFileStream === 'undefined'` rely on it being actually declared as a `class` or a `var` in global scope, while `interface` is only a type entity, so TS will complain that symbol is not defined.

Reply:
Hmm I see. `lib.dom` is defining classes using
```ts
declare var FileSystemFileHandle: {
    prototype: FileSystemFileHandle;
    new(): FileSystemFileHandle;
};
```
and from what I've tried so far this method seems to merge the types without any errors while still allowing `typeof` to work. And this makes it possible for the static `FileSystemDirectoryHandle.getSystemDirectory` to be included again as well.

Reply:
Ah, if they already declare the `var`, then it should be fine.

Reply:
Yeah they do, but I'll declare them here as well to maintain backwards compatibility for ts 4.5

Reply:
Hmm I'm not sure if TypeScript allows merging `var`s too.

Reply:
I thought the same thing but it seems like it does. In my local project I'm looking at
```js
/**
 * @param {FileSystemFileHandle} handle
 */
test(handle) {
  handle.queryPermission();
}
```
which doesn't give any errors, and if I hover over `FileSystemFileHandle` it shows "Available only in secure contexts." from `lib.dom`.